### ISSUE TITLE
A fix for error in tf.layers.conv3d_transpose when inferred batch size

### ISSUE THREAD

**Issue by Joshuaalbert** [tensorflow/tensorflow#15696]
## Context
When using the `tf.layers.conv3d_tranpose` Op with a dynamic batch size and when `use_bias=True` then there is a [well known error that occurs](https://github.com/tensorflow/tensorflow/issues/10520).
The error is due to a [tf.reshape Op that chunks together some of the axes before adding the bias for a slight performance improvement](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/layers/convolutional.py#L1628).
E.g. in the case of `data_format == 'channels_last'`, 
```
outputs_4d = array_ops.reshape(outputs, [
            outputs_shape[0], outputs_shape[1],
            outputs_shape[2] * outputs_shape[3], outputs_shape[4]
        ])
```
gives an error when `outputs_shape[0] == None`, i.e. when batch size is inferred. 

## Simple fix
To fix this with minimal modification to other code, it would be great if someone could replace `outputs_shape[0]` with `-1` in these two lines:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/layers/convolutional.py#L1627
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/layers/convolutional.py#L1632

**Comment by facaiy** [comment]
Similar with #15655 ？

**Comment by Joshuaalbert** [comment]
Yes precisely the same issue.

**Comment by skye** [comment]
@fchollet I see that you're assigned to #10520, are you able to look at this?

**Comment by fchollet** [comment]
Is there an open PR for this? If not, please create one.

**Comment by facaiy** [comment]
@fchollet It seems that conv3d_transpose gets trouble in unknown dimension,  see #15655,  #15572 as well.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by imsheridan** [comment]
This has been fixed by #16307, thus this issue can be closed.
